### ISSUE TITLE
Use Create New Resource menu instead of a long list when creating a new custom resource

### ISSUE THREAD

**Issue by PandaCommando** [godotengine/godot-proposals#3430]
### Describe the project you are working on

I am working on a data-driven game that requires artists to create resources for most of the content. 

### Describe the problem or limitation you are having in your project

![image](https://user-images.githubusercontent.com/21073083/137429398-c2d516f4-50a7-4089-8eac-f1b9f5e7af7a.png)

![image](https://user-images.githubusercontent.com/21073083/137429253-ef4b7be9-96d8-4471-986e-54bff6d90c02.png)

Making new resources is a huge pain because the resource either has to be saved into a separate file or the artists need to scroll down a huge list of all potential resources which takes a lot of time and the artists can't favorite resources that they use often.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The engine already has a very nice new resource selection menu for when you want to create a new resource in memory.
![image](https://user-images.githubusercontent.com/21073083/137429494-8cc20690-9831-4a31-ba82-6bd5517fa12e.png)
![image](https://user-images.githubusercontent.com/21073083/137429524-cfedb687-6d72-4507-989c-d1de68f19817.png)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I propose that instead of listing all possible resources, we just have a button that opens the same resource selector as seen above so we don't have to scroll anymore. This also gives us easier access to the copy/paste controls that are always at the bottom of the list. 

![image](https://user-images.githubusercontent.com/21073083/137430171-1c8f90ee-53dc-4d52-b737-f554152b3c37.png)

The list view is still nice for properties that need a specific resource type like Texture. This button would only show up if the property calls for the generic Resource type.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No. I believe that this will be used very often.

### Is there a reason why this should be core and not an add-on in the asset library?

It deals with the inspector so it must be core.

**Comment by raulsntos** [comment]
Similar to #43

**Comment by PandaCommando** [comment]
> 
> 
> Related to #18 . Overall I agree with refining the current Resource exporting workflow, but I think being able to hint `export` to point to custom classes (like in the above mentioned proposal) would be a better approach.

I agree. I was thinking this could be useful for if a specific type of resource isn't specified. Some of my content management systems can accept textures, audio assets, custom things, etc that don't necessarily all inherit from similar classes so it's useful just to use "Resource".

**Comment by Yagich** [comment]
Ah, I see now. I think my previous reply should be marked off-topic in that case.

I agree with your proposal of unifying the new resource menu and dropdown, but perhaps the dropdown could still list the most commonly used Resource types - or the last few that have been used?
